src/main/java/de/numcodex/feasibility_gui_backend/query/result/ResultService.java

```diff
@@ -0,0 +1,77 @@
+package de.numcodex.feasibility_gui_backend.query.result;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import de.numcodex.feasibility_gui_backend.query.persistence.ResultType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Synchronized;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class ResultService {
+
+  @NonNull
+  private Cache<Long, QueryResult> queryResultCache;
+
+  public QueryResult getResult(Long queryId) {
+    return queryResultCache.getIfPresent(queryId);
+  }
+
+  @Synchronized
+  public void addResultLine(Long queryId, ResultLine resultLine) {
+    String siteName = resultLine.getSiteName();
+    QueryResult queryResult = queryResultCache.getIfPresent(queryId);
+    if (queryResult == null) {
+      queryResult = new QueryResult();
+    } else if (queryResult.getResultsBySite().containsKey(siteName)) {
+      // Only store one result per site - new results get discarded
```

Comment:
Please put this comment in the JavaDoc of this method. This is one important behaviour if this method, callers should know.